datasets/generics_kb/generics_kb.py

```diff
@@ -0,0 +1,239 @@
+title = {GenericsKB: A Knowledge Base of Generic Statements},
+authors={Sumithra Bhakthavatsalam, Chloe Anastasiades, Peter Clark},
+year={2020},
+publisher = {Allen Institute for AI},
+}
+"""
+
+_DESCRIPTION = """\
+The GenericsKB contains 3.4M+ generic sentences about the world, i.e., sentences expressing general truths such as "Dogs bark," and "Trees remove carbon dioxide from the atmosphere." Generics are potentially useful as a knowledge source for AI systems requiring general world knowledge. The GenericsKB is the first large-scale resource containing naturally occurring generic sentences (as opposed to extracted or crowdsourced triples), and is rich in high-quality, general, semantically complete statements. Generics were primarily extracted from three large text sources, namely the Waterloo Corpus, selected parts of Simple Wikipedia, and the ARC Corpus. A filtered, high-quality subset is also available in GenericsKB-Best, containing 1,020,868 sentences. We recommend you start with GenericsKB-Best.
+"""
+
+_HOMEPAGE = "https://allenai.org/data/genericskb"
+
+_LICENSE = "cc-by-4.0"
+
+_FILEPATHS = {
+    "generics_kb_best": "GenericsKB-Best.tsv",
+    "default": "GenericsKB-Best.tsv",
+    "generics_kb": "GenericsKB.tsv",
+    "generics_kb_simplewiki": "GenericsKB-SimpleWiki-With-Context.jsonl",
+    "generics_kb_waterloo": "cskb-waterloo-06-21-with-bert-scores.jsonl",
+}
+
+
+class GenericsKb(datasets.GeneratorBasedBuilder):
+    """ The GenericsKB is the first large-scale resource containing naturally occurring generic sentences, and is rich in high-quality, general, semantically complete statements."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="generics_kb_best",
+            version=VERSION,
+            description="This is the default and recommended config.Comprises of GENERICSKB generics with a score > 0.234 ",
+        ),
+        datasets.BuilderConfig(
+            name="generics_kb", version=VERSION, description="This GENERICSKB that contains 3,433,000 sentences."
+        ),
+        datasets.BuilderConfig(
+            name="generics_kb_simplewiki",
+            version=VERSION,
+            description="SimpleWikipedia is a filtered scrape of SimpleWikipedia pages (simple.wikipedia.org)",
+        ),
+        datasets.BuilderConfig(
+            name="generics_kb_waterloo",
+            version=VERSION,
+            description="The Waterloo corpus is 280GB of English plain text, gathered by Charles Clarke (Univ. Waterloo) using a webcrawler in 2001 from .edu domains.",
+        ),
+    ]
+
+    @property
+    def manual_download_instructions(self):
+        return """\
+      You need to manually download the files needed for the dataset.
+      The <path/to/folder> can e.g. be `~/Downloads/GenericsKB`. Download the following required files from https://drive.google.com/drive/folders/1vqfVXhJXJWuiiXbUa4rZjOgQoJvwZUoT
```

Comment:
Have you tried to use the download_manager with drive urls ?

For example for GenericsKB-Waterloo-With-Context.jsonl.zip
you can use https://drive.google.com/u/0/uc?id=1QLD5RU2vpKWXqg2wofgIJRS_KtdI8ACf&export=download

And more generally you can use the url template
```
https://drive.google.com/u/0/uc?id={id-of-the-file-here}&export=download
```
by replacing `{id-of-the-file}` with the id you get from the link given when you right click > get link on the file in google drive.

Reply:
Hi Quentin,

There are 4 dataset files. The default one Generics KB Best is 90MB. ( It kept failing on my local machine/ did not download with the url you shared.)
The generics KB files is 356MB, the Waterloo context file is 5.5 GB and the simple wiki file is 15 MB.
At the max, we can convert only the simple wiki to download from google drive. Rest all would be a manual download. Do you want me to convert the simple wiki file to download from url, instead of manual download? Please suggest. Thanks.

Reply:
What's the error you got from using the drive download url ?
On my side this works:
```python
from datasets import DownloadManager

# GenericsKB-Best.tsv
url = "https://drive.google.com/u/0/uc?id=12DfIzoWyHIQqssgUgDvz3VG8_ScSh6ng&export=download"                                                        
downloaded_path = DownloadManager().download(url)
```

Reply:
If we're getting errors from those urls we should stick with manual download.
However I'm curious about the error you got and I'd like to see if we can make this work since google drive urls are used by many datasets

Reply:
its working for me too now. I dont remember the error but we discussed on the chat as well. Guess for the Waterloo 1GB can keep manual download, while for the rest will switch to download from url. Will work on this PR and submit my changes.

Reply:
Made the changes. 3 configs now get downloaded from google drive automatically while 1 is a manual download. All tests have passed, updated metadata json and dummy data files. Have pushed all my changes.

Reply:
thanks !